src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/Debug/StartupProjectRegistrar.cs

```diff
@@ -2,150 +2,135 @@
 
 using System;
 using System.ComponentModel.Composition;
+using System.Threading;
 using System.Threading.Tasks;
 using System.Threading.Tasks.Dataflow;
 
 using Microsoft.VisualStudio.ProjectSystem.Debug;
 using Microsoft.VisualStudio.Shell.Interop;
 
+using AsyncServiceProvider = Microsoft.VisualStudio.Shell.AsyncServiceProvider;
+using IAsyncServiceProvider = Microsoft.VisualStudio.Shell.IAsyncServiceProvider;
+
 namespace Microsoft.VisualStudio.ProjectSystem.VS.Debug
 {
     /// <summary>
-    /// <see cref="StartupProjectRegistrar"/> is responsible for adding or removing a project from the Startup list
-    /// depending on whether the active configuration of the a project is debuggable or not.
+    ///     Responsible for adding or removing the project from the startup list based on whether the project
+    ///     is debuggable or not.
     /// </summary>
-    internal class StartupProjectRegistrar : OnceInitializedOnceDisposed
+    internal class StartupProjectRegistrar : OnceInitializedOnceDisposedAsync
     {
+        private readonly IAsyncServiceProvider _serviceProvider;
         private readonly IProjectThreadingService _threadingService;
-        private readonly IActiveConfiguredProjectSubscriptionService _activeConfiguredProjectSubscriptionService;
+        private readonly IActiveConfiguredProjectSubscriptionService _projectSubscriptionService;
         private readonly ActiveConfiguredProject<DebuggerLaunchProviders> _launchProviders;
-        private readonly IVsService<IVsStartupProjectsListService> _startupProjectsListService;
```

Comment:
IVsService enforces the service be accessed on the UI thread - which is not want we want here. EIther I need to remove it and in lieu of IAsyncServiceProvider long term, or change its pattern to "async".